chuffed/vars/bool-view.h

```diff
@@ -75,7 +75,7 @@ class BoolView : public Var {
 
 	bool setVal2(bool x, Reason r = nullptr) const {
 		assert(setValNotR(x));
-		sat.enqueue(getLit(x), r);
+		sat.enqueue(getLit(x), r.pt());
```

Comment:
I'm not sure why the original is incorrect. `SAT::enqueue` accepts a `Reason` argument, by using the `Reason::pt` method, you are always extracting a `Clause*`, independent of what type of `Reason` is constructed. When you then pass it as a `Reason` it would call the constructor of `Reason` again.

This would succeed whenever you have a `Reason` that is a `Clause`, but fails for the other `Reason` variants. It should also trigger an assertion in these cases

Reply:
Got it, sorry... I was getting a build error:
```
In file included from ./chuffed/primitives/primitives.h:71:

./chuffed/vars/bool-view.h:78:28: error: reference to non-static member function must be called; did you mean to call it with no arguments?
   78 |                 sat.enqueue(getLit(x), r.pt);
      |                                        ~~^~
      |                                            ()
1 error generated.
```
and I had to make this change:
```
-               sat.enqueue(getLit(x), r.pt);
+               sat.enqueue(getLit(x), r.pt());
```
Maybe this doesn't transfer back to the mainline.

Reply:
Jip is correct, this needs to be `sat.enqueue(getLit(x), r);` (see commit 7ef8049), which should also fix the issue in your fork of chuffed.